### PR TITLE
Using a new source for each controller to deal with reconciliation

### PR DESCRIPTION
**Pull request by shawn-hurley**
/cc @mhrivnak @fabianvf

### CODE REVIEW

**Comment by fabianvf** [comment]
@shawn-hurley, could you summarize the problem this solves?

File: pkg/controller/controller.go [hunk 1]
```diff
@@ -60,4 +61,11 @@ func Add(mgr manager.Manager, options Options) {
 	if err := c.Watch(&source.Kind{Type: u}, &crthandler.EnqueueRequestForObject{}); err != nil {
 		log.Fatal(err)
 	}
+	r := NewReconcileLoop(time.Duration(time.Minute)*1, options.GVK, mgr.GetClient())
+	cs := &source.Channel{Source: r.Source}
+	cs.InjectStopChannel(r.Stop)
```
Comment (mhrivnak) [inline pkg/controller/controller.go]: From tracing through code and comments, it looks like the controller should do this for us. What happens if you just leave this out?

https://github.com/kubernetes-sigs/controller-runtime/blob/8d93cc75ed7086e85ec7de09332a3bfd8603a959/pkg/runtime/inject/inject.go#L111

I suspect we're not meant to provide this channel or try to use it ourselves.

Comment (mhrivnak) [reply pkg/controller/controller.go]: I think it wants to get this channel: https://github.com/kubernetes-sigs/controller-runtime/blob/8d93cc75ed7086e85ec7de09332a3bfd8603a959/pkg/manager/internal.go#L76

Comment (shawn-hurley) [reply pkg/controller/controller.go]: So you shouldn't have to do anything extra, and your right that is what it is trying to use but https://github.com/kubernetes-sigs/controller-runtime/pull/148 needs to be merged to because it is injecting a nil channel.

Comment (mhrivnak) [reply pkg/controller/controller.go]: If we want to have a stop channel for our reconcile loop, I think we should get it in main.go from signals.SetupSignalHandler() and add it as a field in `github.com/water-hole/ansible-operator/pkg/controller.Options`

**Comment by shawn-hurley** [comment]
PTAL I assume that some names should be changed, but I think the overall structure is not wrong. Thoughts?

File: pkg/controller/source.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+				}
+				for _, u := range ul.Items {
+					e := event.GenericEvent{
+						Meta:   &u,
+						Object: &u,
+					}
+					r.Source <- e
+				}
+			case <-r.Stop:
+				ticker.Stop()
```
Comment (mhrivnak) [inline pkg/controller/source.go]: Maybe consider this for a `defer` statement above the go func.

File: pkg/controller/source.go [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+		Source:   s,
+		GVK:      gvk,
+		Interval: interval,
+		Client:   c,
+	}
+}
+
+// Start - start the reconcile loop
+func (r *ReconcileLoop) Start() {
+	ticker := time.NewTicker(r.Interval)
```
Comment (mhrivnak) [inline pkg/controller/source.go]: Any reason to not scope this inside the go func?
